examples/vue-ethers5/src/App.vue

```diff
@@ -1,84 +1,139 @@
 <script lang="ts" setup>
-import { ref, onMounted, watch } from 'vue'
-import { arbitrum, mainnet } from '@reown/appkit/networks'
+import { ref, onMounted } from 'vue'
+import { mainnet, polygon, base } from '@reown/appkit/networks'
 import { Ethers5Adapter } from '@reown/appkit-adapter-ethers5'
-import {
-  createAppKit,
-  useAppKit,
-  useAppKitEvents,
-  useAppKitState,
-  useAppKitTheme
-} from '@reown/appkit/vue'
+import { createAppKit, useAppKitEvents, useAppKitState, useAppKitTheme } from '@reown/appkit/vue'
 
-const error = ref('')
-
-// 1. Get projectId
 const projectId = import.meta.env.VITE_PROJECT_ID
 if (!projectId) {
   throw new Error('VITE_PROJECT_ID is not set')
 }
 
-// 2. Create ethers adapter
+// Initialize Ethers5 adapter
 const ethersAdapter = new Ethers5Adapter()
 
-// 3. Create modal
-createAppKit({
+// Initialize AppKit
+const modal = createAppKit({
   adapters: [ethersAdapter],
-  networks: [mainnet, arbitrum],
+  networks: [mainnet, polygon, base],
   projectId,
   metadata: {
     name: 'AppKit Vue Ethers Example',
     description: 'AppKit Vue Ethers Example',
-    url: '',
-    icons: []
+    url: 'https://reown.com/appkit',
+    icons: ['https://avatars.githubusercontent.com/u/179229932?s=200&v=4']
   }
 })
 
-const modal = useAppKit()
-const state = useAppKitState()
-const { setThemeMode, themeMode, themeVariables } = useAppKitTheme()
+// State Management
+const accountState = ref({})
+const networkState = ref({})
+const appState = useAppKitState()
+const { setThemeMode } = useAppKitTheme()
 const events = useAppKitEvents()
+const walletInfo = ref({})
+const themeState = ref({ themeMode: 'light', themeVariables: {} })
 
+// Theme toggle function
 const toggleTheme = () => {
-  const newTheme = themeMode.value === 'dark' ? 'light' : 'dark'
+  const newTheme = themeState.value.themeMode === 'dark' ? 'light' : 'dark'
   setThemeMode(newTheme)
-}
-
-// Watch for theme changes and update body class
-watch(themeMode, newTheme => {
+  themeState.value.themeMode = newTheme
   document.body.className = newTheme
-})
+}
 
-// Set initial theme class on mount
+// Subscriptions
 onMounted(() => {
-  document.body.className = themeMode.value
+  // Set initial theme
+  document.body.className = themeState.value.themeMode
+
+  // Setup subscriptions
+  modal.subscribeAccount(state => {
+    accountState.value = state
+  })
+
+  modal.subscribeNetwork(state => {
+    networkState.value = state
+  })
+
+  modal.subscribeTheme(state => {
+    themeState.value = state
+    document.body.className = state.themeMode
+  })
+
+  modal.subscribeWalletInfo(state => {
+    // @ts-ignore
```

Comment:
Needed ?

Reply:
On the Vue the TS extension was not working fine, didn't want to spend time resolving it. It's sample app related thing, not the AppKit itself, so safe to ignore for now